specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2022-04-01-preview/apimauthorizationproviders.json

```diff
@@ -0,0 +1,805 @@
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/AuthorizationIdParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/AuthorizationAccessPolicyIdParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The response body contains the specified authorization access policy entity.",
+            "schema": {
+              "$ref": "./definitions.json#/definitions/AuthorizationAccessPolicyContract"
+            },
+            "headers": {
+              "ETag": {
+                "description": "Current entity state version. Should be treated as opaque and used to make conditional HTTP requests.",
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./apimanagement.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "AuthorizationAccessPolicy"
+        ],
+        "operationId": "AuthorizationAccessPolicy_CreateOrUpdate",
+        "description": "Creates or updates Authorization Access Policy.",
+        "x-ms-examples": {
+          "ApiManagementCreateAuthorizationAccessPolicy": {
+            "$ref": "./examples/ApiManagementCreateAuthorizationAccessPolicy.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "./apimanagement.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ServiceNameParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/AuthorizationProviderIdParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/AuthorizationIdParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/AuthorizationAccessPolicyIdParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "./definitions.json#/definitions/AuthorizationAccessPolicyContract"
+            },
+            "description": "Create parameters."
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/IfMatchOptionalParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
```

Comment:
@LFZ96 can you check this comment

Reply:
@TimLovellSmith Similar to above, we do support Creation. Should I change the status to be 201 instead of 200?

Reply:
Create operation is generally really 'create or update' and can either/both 200 and 201.

Reply:
A side note: PUT is generally create or replace and PATCH is create or update. In HTTP, a PUT over an existing resource is the logical equivalent of a DELETE/PUT. We're trying to make this (PUT=create or replace) be the culture in Azure (since it is the culture for HTTP). It matters a lot because of the way Azure does its versioning.